### PR TITLE
[CAMEL-19100] Fixes lost dataChangeFilterTrigger in camel-milo

### PR DESCRIPTION
**Pull request by rangoy**
# Description

Milo component does not use dataChangeFilterTrigger value from route, this was replaced by StatusValueTimestamp during setup.

Ref https://issues.apache.org/jira/browse/CAMEL-19100

# Target

Targets 3.x branch as camel-milo is disabled in 4.x branch, waiting for jakarata update of eclipse milo, ref https://github.com/eclipse/milo/issues/1098

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I formatted the code using `mvn -Pformat,fastinstall install && mvn -Psourcecheck`

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by rangoy** [comment]
I see that I might have chosen the wrong target branch. Should I change it to 3.x?

**Comment by essobedo** [comment]
> I see that I might have chosen the wrong target branch. Should I change it to 3.x?

3.20.x is to target 3.20.3, 3.x is to target 3.21.0 and main is to target 4.0-M2

File: components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MonitorFilterConfiguration.java [hunk 1]
```diff
@@ -80,7 +80,9 @@ public MonitoringFilter createMonitoringFilter() {
         switch (monitorFilterType) {
             default:
                 return new DataChangeFilter(
-                        DataChangeTrigger.StatusValueTimestamp, this.getDataChangeFilterDeadbandType(),
+                        this.dataChangeFilterTrigger != null
+                                ? this.dataChangeFilterTrigger : DataChangeTrigger.StatusValueTimestamp,
```
Comment (essobedo) [inline components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MonitorFilterConfiguration.java:84]: Warning this is a behavior change, as far as I can see the current default value is [`DataChangeTrigger.StatusValue`](https://github.com/apache/camel/blob/c1a8be35640603c35160aba07d715c3cd3878af0/components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MonitorFilterConfiguration.java#L41) whereas at runtime it used `DataChangeTrigger.StatusValueTimestamp`. Maybe should change the default value to `DataChangeTrigger.StatusValueTimestamp` and use `dataChangeFilterTrigger` here.

Comment (rangoy) [reply components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MonitorFilterConfiguration.java]: Good point. I did not pay attenation to that the filter had a default different from what was actually applied here. 

I've updated the pullrequest to use the old effective default value (StatusValueTimestamp)

File: components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MonitorFilterConfiguration.java [hunk 2]
```diff
@@ -37,8 +37,8 @@ public class MonitorFilterConfiguration implements Cloneable {
     @UriParam(defaultValue = "0.0", description = "Deadband value for MonitorFilterType DataChangeFilter.")
     private Double dataChangeFilterDeadbandValue = 0.0;
 
-    @UriParam(description = "Daata change trigger for data change monitor filter type.")
-    private DataChangeTrigger dataChangeFilterTrigger = DataChangeTrigger.StatusValue;
+    @UriParam(description = "Data change trigger for data change monitor filter type.")
```
Comment (essobedo) [inline components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MonitorFilterConfiguration.java:40]:
```suggestion
    @UriParam(defaultValue = "StatusValueTimestamp", description = "Data change trigger for data change monitor filter type.")

```
Let's add the default value to the metadata too

File: components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MonitorFilterConfiguration.java (hunk 1 above)
Comment (essobedo) [reply components/camel-milo/src/main/java/org/apache/camel/component/milo/client/MonitorFilterConfiguration.java:84]: For the sake of consistency with the rest of the code, don't check if it is `null` only call the corresponding getter

**Review by essobedo — APPROVED** [review]

**Comment by essobedo** [comment]
@rangoy Could you please propose also a PR for the branch 3.x?

**Comment by rangoy** [comment]
> @rangoy Could you please propose also a PR for the branch 3.x?

Sure. From my fork?

**Comment by essobedo** [comment]
Yes please

**Pull request merged**